src/screens/assetDetails/children/coinSummary.tsx

```diff
@@ -22,12 +22,12 @@ export const CoinSummary = ({
   onActionPress,
   onInfoPress,
 }: CoinSummaryProps) => {
-  const { balance, estimateValue, savings, extraDataPairs, actions } = coinData;
+  const { balance, estimateValue, savings, extraDataPairs, actions, precision } = coinData;
 
   const valuePairs = [
     {
       dataKey: 'amount_desc',
-      value: balance,
+      value: balance.toFixed(precision),
```

Comment:
@aliseyalvi are we setting any default value for precision, how are we managing non engine precision here.?

Reply:
@noumantahir the above change only affects engine tokens and we are considering all engine tokens returns precision. We can add default precision here as well to handle null case

Reply:
@noumantahir added default precision to 3 in coinSummary
https://github.com/ecency/ecency-mobile/pull/2624/commits/f82eedd9d0add26effa6ba3154996175585a75d3